src/functionHandlers/revokePdtHealthCert/handler.ts

```diff
@@ -0,0 +1,213 @@
+      data.issuers[0].revocation?.location === undefined
+    ) {
+      throw new CodedError(
+        "INVALID_DOCUMENT",
+        `Unable to revoke certificate - revocation fields missing in certificate`
+      );
+    }
+
+    /* 3. Extract out clinic endorsed cert (attachment) from MOH endorsed cert (given) */
+    const encodedPreEndorsedHealthcert = data.attachments
+      ? data.attachments[0].data
+      : undefined;
+
+    if (encodedPreEndorsedHealthcert === undefined) {
+      throw new CodedError(
+        "INVALID_DOCUMENT",
+        `Unable to extract pre-endorsed HealthCert from given certificate`
+      );
+    }
+
+    const decodedPreEndorsedHealthCert = Buffer.from(
+      encodedPreEndorsedHealthcert,
+      "base64"
+    ).toString("ascii");
+
+    const preEndorsedHealthCert = getData(
+      JSON.parse(
+        decodedPreEndorsedHealthCert
+      ) as WrappedDocument<PDTHealthCertV2>
+    );
+
+    /* 4. Revoke cert if caller is indeed the provider of the cert */
+    const healthCertClinicDomain =
+      preEndorsedHealthCert.issuers[0].identityProof?.location;
+
+    if (healthCertClinicDomain === undefined) {
+      throw new CodedError(
+        "INVALID_DOCUMENT",
+        "Unable to revoke certificate - missing domain in identity proof"
+      );
+    }
+
+    const caller =
+      providerApiKey !== undefined ? getProvider(providerApiKey) : undefined;
+
+    if (caller === undefined) {
+      throw new CodedError(
+        "INVALID_API_KEY",
+        `Unable to revoke certificate - key is invalid`
+      );
+    }
+
+    if (caller?.domain === undefined) {
+      throw new CodedError(
+        "MISSING_CLINIC_DOMAIN_ERROR",
+        `Unable to revoke certificate - unable to find domain`
+      );
+    }
+
+    if (!healthCertClinicDomain.includes(caller.domain)) {
+      throw new CodedError(
+        "INVALID_PROVIDER",
+        `Unable to revoke certificate - caller clinic must match provider clinic in certificate`,
+        `callerDomain !== clinicDomain`
+      );
+    }
+
+    let result: RevocationResult;
+    try {
+      result = await revokePdtHealthcert(reference, wrappedDocument);
+    } catch (e) {
+      throw e instanceof CodedError
+        ? e
+        : new CodedError(
+            "REVOKE_PDT_ERROR",
+            "Unable to revoke document",
+            JSON.stringify(serializeError(e))
+          );
+    }
+
+    // Log healthcert reason and revocation of cert
+    if (hcReasonCode !== undefined) {
+      const hcReason =
+        config.hcProviderReasonCodes[
+          hcReasonCode as keyof typeof config.hcProviderReasonCodes
+        ];
+
+      if (hcReason === undefined) {
+        throw new CodedError(
+          "INVALID_REVOCATION_REASON_CODE",
+          "Unable to revoke document - invalid reason code"
+        );
+      }
+
+      traceWithRef(
+        `certificate ${wrappedDocument.signature.targetHash} revoked success ${hcReason}`
```

Comment:
@yapyuyou Is it sufficient to log `targetHash` and the `hcReason` provided by the HC provider?

Reply:
yup it's fine